### PR TITLE
[vcpkg ci,liburing] Improve license identifier check

### PR DESCRIPTION
**Pull request by dg0yt**
- Add annotations directly below `license` field, for better visibility.
- Write to workflow summary.
- Handle more complex `license` expressions.

**Screenshot:**

screenshot

### CODE REVIEW

File: .github/workflows/untrustedPR.yml [hunk 1]
```diff
@@ -149,23 +157,28 @@ jobs:
...
-              }
-              if (deprecated_license !== "") {
-                license_output += 'If you feel able to do so, please consider replacing the deprecated license identifiers in the following files:\n'
-                license_output += deprecated_license
-                license_output += "\n\nDeprecated and non deprecated license identifiers can be found at https://spdx.org/licenses/#deprecated\n"
-              }
+            if (missing_license !== "" || deprecated_license.length > 0) {
+              core.summary.addRaw('You have modified or added at least one vcpkg.json where you should check the \"license\" field.');
+              core.summary.addEOL();
+            }
```
Comment (dg0yt) [inline .github/workflows/untrustedPR.yml:163]: Shoudn't I see an effect?

**Review by FrankXie05 — DISMISSED** [review]

File: .github/workflows/untrustedPR.yml [hunk 2]
```diff
@@ -149,23 +157,28 @@ jobs:
...
 
+            if (missing_license !== "") {
+              license_output = 'If you feel able to do so, please consider adding a "license" field to the following files:\n'
+              license_output += missing_license
+              license_output += "\n\nValid values for the license field can be found at https://learn.microsoft.com/en-us/vcpkg/reference/vcpkg-json#license\n\n"
```
Comment (BillyONeal) [inline .github/workflows/untrustedPR.yml:168]:
```suggestion
              license_output += "\n\nValid values for the license field can be found at https://learn.microsoft.com/vcpkg/reference/vcpkg-json#license\n\n"

```

File: .github/workflows/untrustedPR.yml [hunk 3]
```diff
@@ -149,23 +157,28 @@ jobs:
...
               core.warning(license_output);
             }
 
+            for (let line of deprecated_license) {
+              [file, line_number, match] = line.split(':');
+              var license_output = `The license identifier ${match} is deprecated. `;
```
Comment (BillyONeal) [inline .github/workflows/untrustedPR.yml:174]:
```suggestion
              var license_output = `The license identifier ${match} is deprecated in favor of another license identifier. For example, LGPL2.1 was deprecated in favor of LGPL2.1-Only or LGPL2.1-Or-Later `;
```

Or similar; as written it sounds like we're deprecating SDPX license identifiers

**Comment by JavierMatosD** [comment]
We will take this subject to the nitpicks :)

File: .github/workflows/untrustedPR.yml (hunk 3 above)
Comment (dg0yt) [reply .github/workflows/untrustedPR.yml:174]:
```suggestion
              var license_output = `SPDX deprecated license identifier ${match}. Please check https://spdx.org/licenses/ for an updated identifier.`;

```

File: .github/workflows/untrustedPR.yml [hunk 4]
```diff
@@ -149,23 +157,30 @@ jobs:
...
 
               core.warning(license_output);
             }
 
+            for (let line of deprecated_license) {
+              [file, line_number, match] = line.split(':');
+              var license_output = `SPDX deprecated license identifier ${match}. Please check https://spdx.org/licenses/ for an updated identifier.`;
+
+              license_output += "If you feel able to do so, please consider replacing it.\n"
+              license_output += "License identifiers can be found at https://spdx.org/licenses/."
```
Comment (dg0yt) [inline .github/workflows/untrustedPR.yml:178]:
```suggestion
              var license_output = `SPDX deprecated license identifier ${match}.`;
              license_output += " If you feel able to do so, please consider replacing it."
              license_output += " Updated license identifiers can be found at https://spdx.org/licenses/."

```

File: .github/workflows/untrustedPR.yml [hunk 5]
```diff
@@ -149,23 +157,30 @@ jobs:
...
+              license_output += "\n\nValid values for the license field can be found at https://learn.microsoft.com/vcpkg/reference/vcpkg-json#license\n\n"
 
               core.warning(license_output);
             }
 
+            for (let line of deprecated_license) {
+              [file, line_number, match] = line.split(':');
+              var license_output = `SPDX deprecated license identifier ${match}.`;
+              license_output += " If you feel able to do so, please consider replacing it."
+              license_output += " Updated license identifiers can be found at https://spdx.org/licenses/."
```
Comment (dg0yt) [inline .github/workflows/untrustedPR.yml:177]:
```suggestion
              var license_output = `SPDX deprecated license identifier ${match}.`;
              license_output += "\nIf you feel able to do so, please consider replacing it."
              license_output += "\nUpdated license identifiers can be found at https://spdx.org/licenses/."

```

**Pull request merged**